src/invidious/user/imports.cr

```diff
@@ -290,42 +290,38 @@ struct Invidious::User
     end
 
     def from_newpipe(user : User, body : String) : Bool
-      io = IO::Memory.new(body)
-
-      Compress::Zip::File.open(io) do |file|
+      Compress::Zip::File.open(IO::Memory.new(body), true) do |file|
         file.entries.each do |entry|
           entry.open do |file_io|
-            # Ensure max size of 4MB
-            io_sized = IO::Sized.new(file_io, 0x400000)
-
             next if entry.filename != "newpipe.db"
 
-            tempfile = File.tempfile(".db")
-
-            begin
-              File.write(tempfile.path, io_sized.gets_to_end)
-            rescue
-              return false
-            end
-
-            db = DB.open("sqlite3://" + tempfile.path)
+            # Ensure max size of 4MB
+            io_sized = IO::Sized.new(file_io, 0x400000)
 
-            user.watched += db.query_all("SELECT url FROM streams", as: String)
-              .map(&.lchop("https://www.youtube.com/watch?v="))
+            temp = File.tempfile(".db") do |tempfile|
```

Comment:
Would we maybe want to randomly generate part of the file file's name? (ex: .db-{UUID}) I imagine this might cause some issues if invidious starts to use Concurrency

Reply:
Yes, that might be a huge problem! The ideal option would be ti use in-memory databases and use [`sqlite3_serialize`](https://www.sqlite.org/c3ref/serialize.html), but that requires to make a PR to [crystal-sqlite3](https://github.com/crystal-lang/crystal-sqlite3)

Reply:
Yes, that would be even better! I imagine there might need to be a PR made here as well (if serialization is supported by other databases): [crystal-db](https://github.com/crystal-lang/crystal-db)

Reply:
In memory-databases are already supported in crystal-sqlite3 like so:

```crystal
DB.open "sqlite3::memory" do | db |
  db.exec "create table contacts (name text, age integer)"
  db.exec "insert into contacts values (?, ?)", "John Doe", 30
end
```

The only thing that needs to be added is a way to actually access an in-memory database.

There's no need for a PR directly imo, an issue would be enough.

Reply:
Done: https://github.com/crystal-lang/crystal-sqlite3/issues/95